framework/org.eclipse.concierge/test/org/eclipse/concierge/BundleGetDataFileTest.java

```diff
@@ -55,6 +55,17 @@ private void setupDefaultBundles() throws Exception {
 		builder.addManifestHeader("Fragment-Host", "bundle");
 		fragmentUnderTest = installBundle(builder);
 	}
+	private static String joinString(String delimiter, String... args){
+		StringBuilder sb = new StringBuilder();
+		int i = 0;
+		for( ; i < args.length -1; i++){
```

Comment:
Why not writing: `for (int i = 0 ; i < args.length - 1 ; i++ ) {`

I did not run the code yet: Does it not need to be `args.length - 2`? As you will add the last argument out of the loop, which is correct to not put the separator at the end.

Reply:
Because for access to last argument by `i` at out of loop.

For example, `joinPath` called with 3 strings. (`args.length` is `3`).
+ 1st(`i = 0`): `sb` became `"1" + "/"` 
+ 2nd(`i = 1`): `sb` became `"1" + "/" + "2" + "/"`
+ 3rd(`i = 2`): Not execution. (`i < args.length -1` => `2 < 2` => `false`)
+ Out of loop(`i = 2`): `sb` became `"1" + "/" + "2" + "/" + "3"`

So, this for-loop condition is `i < args.length -1`.

I hope I can explain it well..